### PR TITLE
refine parameter checking framework

### PR DESCRIPTION
**Pull request by wanghouqi**
refine payload parameter check

test: passed the python test

Signed-off-by: wanghouqi <wanghouqi@vip.qq.com>

### CODE REVIEW

File: test/security_test/createkey.py [hunk 1]
```diff
@@ -26,14 +26,14 @@ def init_params(appid, apikey, timestamp_id, keyspec, origin, sign=None):
     payload["origin"] = origin
     params = OrderedDict()
     params["appid"] = appid
-    params["payload"] = urllib.parse.unquote(urllib.parse.urlencode(payload))
+    params["payload"] = urllib.parse.unquote_plus(urllib.parse.urlencode(payload))
```
Comment (yang8621) [inline test/security_test/createkey.py:29]: Why use unquote_plus? The difference is that unquote_plus also changes + to space.

File: ehsm_kms_service/secret_manager_apis.js [hunk 2]
```diff
@@ -133,25 +67,20 @@ const get_SM_Masterkey = async (DB, appid, encryptionKeyId) => {
             ret.sm_masterKey = res.docs[0].sm_default_cmk
         }
     } else if (typeof (encryptionKeyId) == 'string') {
-        // query SM_MasterKey by encryptionKeyId
-        if (check_encryptionKeyId_format(encryptionKeyId)) {
```
Comment (yang8621) [inline ehsm_kms_service/secret_manager_apis.js:137]: no need to check the format anymore?

Comment (wanghouqi) [reply ehsm_kms_service/secret_manager_apis.js:137]: all check moved to payload_checker.js

File: test/security_test/createkey.py (hunk 1 above)
Comment (wanghouqi) [reply test/security_test/createkey.py:29]: unquote_plus can fix the space sign error.

Comment (yang8621) [reply test/security_test/createkey.py:29]: is there an example to show the error we met?

Comment (wanghouqi) [reply test/security_test/createkey.py:29]: on secretData, if you put "test data 1", on unquote will be sign error.

File: ehsm_kms_service/constant.js [hunk 3]
```diff
@@ -1,43 +1,33 @@
...
-const SECRETDATA_LENGTH_MAX = 4 * 1024
-const DESCRIPTION_LENGTH_MAX = 4 * 1024
-const DEFAULT_DELETE_RECOVERY_DAYS = 30
-const QUOTE_LENGTH_MAX = 10924 // the length after 8K base64
-const NONCE_LENGTH_MAX = 4 * 1024
-const CHALLENGE_LENGTH_MAX = 4 * 1024
-const UUID_LENGTH = 36
-const MR_ENCLAVE_LENGTH = 64
-const MR_SIGNER_LENGTH = 64
+const Definition = {
```
Comment (syan10) [inline ehsm_kms_service/constant.js:1]: “Definition” is not a good name!

Comment (wanghouqi) [reply ehsm_kms_service/constant.js:1]: What is your desired name? eg. DATA_DICT ?

File: ehsm_kms_service/function.js [hunk 4]
```diff
@@ -325,8 +304,9 @@ const params_sort_str = (sign_params) => {
     }
     return str
   } catch (error) {
-    res.send(_result(404, 'Not Found', {}))
+    console.info('params_sort_str :: ', error)
```
Comment (syan10) [inline ehsm_kms_service/function.js:307]: what does it mean "params_sort_str"?

**Review by yang8621 — APPROVED** [review]

**Review by HawieLong — APPROVED** [review]

File: ehsm_kms_service/function.js (hunk 4 above)
Comment (wanghouqi) [reply ehsm_kms_service/function.js]: the params_sort_str is funcion name, it will be remove and change to user logger.

**Review by syan10 — APPROVED** [review]

**Pull request merged**